microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -155,79 +134,52 @@ func reconcileResource(ctx context.Context, dynamicClient *dynamic.DynamicClient
 		if res == "" || !strings.Contains(res, "kind:") {
 			continue
 		}
-		// if create failed, wait 2s and retry, this will be removed when the monitor task is implemented
-		for {
-			createdObj, err := applyResourceToK8s(ctx, dynamicClient, ns, []byte(res))
-			if err != nil {
-				fmt.Printf("Failed to reconcile resource: %v\n", err)
-			} else {
-				fmt.Printf("Success to reconcile %s: %s\n", createdObj.GetKind(), createdObj.GetName())
+		createdObj, err := applyResourceToK8s(ctx, client, ns, svc, []byte(res))
+		if err != nil {
+			fmt.Printf("Failed to reconcile resource: %v\n", err)
+		} else {
+			fmt.Printf("Success to reconcile %s: %s\n", createdObj.GetKind(), createdObj.GetName())
 
-				// return the service obj to get the serivce URL from it
-				if retSvc != nil && createdObj.GetKind() == "Service" {
-					err = scheme.Scheme.Convert(createdObj, retSvc, nil)
-					if err != nil {
-						fmt.Printf("Failed to save service: %v\n", err)
-					}
+			// return the service obj to get the serivce URL from it
+			if retSvc != nil && createdObj.GetKind() == Service {
+				err = scheme.Scheme.Convert(createdObj, retSvc, nil)
+				if err != nil {
+					fmt.Printf("Failed to save service: %v\n", err)
 				}
+			}
 
-				if createdObj.GetKind() == "Deployment" {
-					var newEnvVars []corev1.EnvVar
-					if svcCfg != nil {
-						for name, value := range *svcCfg {
-							if name == "endpoint" {
-								continue
-							}
-							itemEnvVar := corev1.EnvVar{
-								Name:  name,
-								Value: value,
-							}
-							newEnvVars = append(newEnvVars, itemEnvVar)
+			if createdObj.GetKind() == Deployment && step != Router {
+				var newEnvVars []corev1.EnvVar
+				if svcCfg != nil {
+					for name, value := range *svcCfg {
+						if name == "endpoint" {
+							continue
 						}
-					}
-					if len(newEnvVars) > 0 {
-						deployment := &appsv1.Deployment{}
-						err = runtime.DefaultUnstructuredConverter.FromUnstructured(createdObj.UnstructuredContent(), deployment)
-						if err != nil {
-							fmt.Printf("Failed to save deployment: %v\n", err)
-						}
-						for i := range deployment.Spec.Template.Spec.Containers {
-							deployment.Spec.Template.Spec.Containers[i].Env = append(
-								deployment.Spec.Template.Spec.Containers[i].Env,
-								newEnvVars...)
-						}
-						modifiedObj, derr := runtime.DefaultUnstructuredConverter.ToUnstructured(deployment)
-						if derr != nil {
-							fmt.Printf("Failed to marshal updated deployment: %v", derr)
-						}
-
-						// Remove managedFields from the unstructured object
-						if _, ok := modifiedObj["metadata"].(map[string]interface{}); ok {
```

Comment:
I suggest that you need to test that the environment variables has bees set if you remove these lines.

Reply:
done that, thanks for reminding me